cmd/frontend/frontend.go

```diff
@@ -0,0 +1,172 @@
+package main
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"net"
+	"net/http"
+	"os"
+	"sync/atomic"
+
+	"github.com/Azure/ARO-HCP/pkg/api/arm"
+)
+
+const (
+	// Literal path segments must be lowercase because
+	// MiddlewareLowercase converts the request URL to
+	// lowercase before multiplexing.
+	ResourceProviderNamespace = "microsoft.redhatopenshift"
+
+	SubscriptionPath  = "/subscriptions/{subscriptionId}"
+	ResourceGroupPath = SubscriptionPath + "/resourcegroups/{resourceGroupName}"
+	ResourceTypePath  = ResourceGroupPath + "/" + ResourceProviderNamespace + "/{resourceType}"
+	ResourceNamePath  = ResourceTypePath + "/{resourceName}"
+)
+
+type PathValues struct {
```

Comment:
At some point we'll have to support a child resource type, but good for now.  

Also, it may make sense to use the parseResourceId functions that come out of the box with Azure SDK for go.

Reply:
Good call.  Not gonna import azure-sdk-for-go in this PR, but I'll take it under advisement.